### PR TITLE
Add a few more tests to mutators

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR:

- [x] Adds a few more tests for mutators

### CODE REVIEW

File: tests/Fixtures/Autoloaded/PublicVisibility/pv-protected-parent.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+abstract class SameAbstract
+{
+    protected abstract function foo();
+}
+class Child extends SameAbstract
+{
+    public function foo()
+    {
+    }
+}
```
Comment (BackEndTea) [inline tests/Fixtures/Autoloaded/PublicVisibility/pv-protected-parent.php]: Note to self: missing new line at end of file

File: tests/Mutator/Number/DecrementIntegerTest.php [hunk 2]
```diff
@@ -303,6 +303,39 @@ public function provideMutationCases(): array
...
+$foo = 0;
+PHP
+                ,
+                <<<'PHP'
+<?php
+
+$foo = -1;
+PHP
+            ],
+            'It does not decrement an assignment of 1' => [
```
Comment (borNfreee) [inline tests/Mutator/Number/DecrementIntegerTest.php]: could you please remind why don't we decrement `1`?

Comment (BackEndTea) [reply tests/Mutator/Number/DecrementIntegerTest.php]: Because of the `OneZeroInteger` mutator, which does exactly that.

Comment (borNfreee) [reply tests/Mutator/Number/DecrementIntegerTest.php]: ah, thanks, probably worths adding a comment before `if (!$node instanceof Node\Scalar\LNumber || $node->value === 1) {`

**Comment by borNfreee** [comment]
Thank you @BackEndTea

**Pull request merged**